tuple-table/src/main/java/com/iluwatar/tupletable/MemberTupleDao.java

```diff
@@ -0,0 +1,149 @@
+  /**
+   * Method implemented to find the member based on the input param.
+   *
+   * @param memberNo is key with which tuple record is looked up
+   * @return member retrieved from the database
+   */
+  public MemberDto findMember(long memberNo) throws SQLException, ClassNotFoundException {
+    Database db = new Database();
+    MemberDto member = new MemberDto();
+    member.setMemberNumber(memberNo);
+
+    try (Connection con = db.getConnection();
+         PreparedStatement ps = con.prepareStatement("select fieldname, numerical, string "
+                 + "from object_data where obj_pk = ?")) {
+      // create a new table
+      ps.setLong(1, memberNo);
+      try (ResultSet rs = ps.executeQuery()) {
+        while (rs.next()) {
+          String fieldName = rs.getString(1);
+          String strVal = rs.getString(3);
+          if (strVal != null) {
+            setVal(fieldName, member, strVal);
+          } else {
+            //we do this indirectly to make database typecasting more reliable
+            long lngVal = rs.getLong(2);
+            if (!rs.wasNull()) {
+              setVal(fieldName, member, lngVal);
+            }
+          }
+        }
+      }
+    }
+    return member;
+  }
+
+  /**
+   * Method to create the table if not exist (for demo purpose).
+   *
+   * @should create the object_data table when called
+   */
+  public void createTableIfNotExists() throws SQLException, ClassNotFoundException {
+    String sql = "CREATE TABLE IF NOT EXISTS object_data (\n"
+            + "OBJ_PK int NOT NULL,\n"
+            + "FIELDNAME varchar(20) NOT NULL,\n"
+            + "NUMERICAL int,\n"
+            + "STRING varchar(255)\n"
+            + ");";
+    Database db = new Database();
+    try (Connection con = db.getConnection();
+         Statement stmt = con.createStatement()) {
+      // create a new table
+      stmt.execute(sql);
+    }
+  }
+
+  /**
+   * Method implemented to save the member in database.
+   *
+   * @param member object
+   * @should save member in the DB
+   */
+  public void saveMember(MemberDto member) throws SQLException, ClassNotFoundException,
+          InvocationTargetException, IllegalAccessException {
+    Database db = new Database();
+    long memberNo = member.getMemberNumber();
+    if (memberNo >= 1) {
+      try (Connection con = db.getConnection()) {
+        // create a new table
+        try (PreparedStatement ps = con.prepareStatement(
+                "delete from object_data where obj_pk = ?");) {
+          ps.setLong(1, memberNo);
+          ps.executeUpdate();
+          try (PreparedStatement ps1 = con.prepareStatement(
+                  "insert into object_data (obj_pk, fieldname, "
+                          + "numerical, string) values (?,?,?,?);");
+          ) {
+            ps1.setLong(1, memberNo);
+            extracted(member, ps1);
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Extracted method for saveMember method.
```

Comment:
The method name should accurately describe what it does. `Extracted` is a strange choice for a name...

Reply:
provided a meaningful name